JSONObject.java

```diff
@@ -1797,7 +1856,7 @@ public static Object wrap(Object object) {
             }
             if (object instanceof Map) {
                 Map<?, ?> map = (Map<?, ?>) object;
-                return new JSONObject(map);
+                return new JSONObject(map, useJavaNullAsJsonNull);
```

Comment:
you should also update lines 1852 and 1855 to use the new JSONArray constructors:
```java
if (object instanceof Collection) {
                 Collection<?> coll = (Collection<?>) object;
                 return new JSONArray(coll, useJavaNullAsJsonNull);
             }
             if (object.getClass().isArray()) {
                 return new JSONArray(object, useJavaNullAsJsonNull);
              }
```